AmplifyPlugins/API/AWSAPICategoryPlugin/Core/AppSyncListPayload.swift

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSPluginsCore
+
+/// Store metadata about the request alongside AppSync's GraphQL response for a List operation, useful for creating
+/// `AppSyncListProvider` that is capable of performing pagination based on the metadata.
+public struct AppSyncListPayload: Codable {
+    private let variables: [String: JSONValue]?
+
+    let graphQLData: JSONValue
+    let apiName: String?
+
+    public init(graphQLData: JSONValue,
+                apiName: String?,
+                variables: [String: JSONValue]?) {
+        self.apiName = apiName
+        self.variables = variables
+        self.graphQLData = graphQLData
+    }
+
+    /// Extract from the `variables` object the original GraphQL "filter" is required to perform pagination by
+    /// the `AppSyncListProvider` based on the original request filter criteria that was passed in as
+    /// a `QueryPredicate`. This may be the association data used if lazy loading was originally performed, or the
+    /// predicate that was passed in when retrieving a list (`Amplify.API.query(.paginatedList(where: predicate)`)
+    var graphQLFilter: [String: Any]? {
+        guard let storedVariables = variables,
+           let filters = storedVariables["filter"],
+           case let .object(filterValue) = filters else {
+            return nil
+        }
```

Comment:
updated to multiple guard statements

previous conversation: https://github.com/aws-amplify/amplify-ios/pull/990/files#r548597651